pulumi-eks-go/main.go

```diff
@@ -0,0 +1,439 @@
+		// Docs: https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Route_Tables.html
+
+		// Private Route Table for Private Subnets
+		privateRouteTable, err := ec2.NewRouteTable(ctx, prefix+"-rtb-private-1", &ec2.RouteTableArgs{
+			VpcId: vpc.ID(),
+			Routes: ec2.RouteTableRouteArray{
+				&ec2.RouteTableRouteArgs{
+					// To Internet via NAT
+					CidrBlock: pulumi.String("0.0.0.0/0"),
+					GatewayId: natGw1.ID(),
+				},
+			},
+			Tags: pulumi.StringMap{
+				"Name": pulumi.String(prefix + "-rtb-private-1"),
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		// Public Route Table for Public Subnets
+		publicRouteTable, err := ec2.NewRouteTable(ctx, prefix+"-rtb-public-1", &ec2.RouteTableArgs{
+			VpcId: vpc.ID(),
+			Routes: ec2.RouteTableRouteArray{
+				// To Internet via IGW
+				&ec2.RouteTableRouteArgs{
+					CidrBlock: pulumi.String("0.0.0.0/0"),
+					GatewayId: igw1.ID(),
+				},
+			},
+			Tags: pulumi.StringMap{
+				"Name": pulumi.String(prefix + "-rtb-public-1"),
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		// Associate Private Subs with Private Route Tables
+		_, err = ec2.NewRouteTableAssociation(ctx, prefix+"-rtb-assoc-priv-1", &ec2.RouteTableAssociationArgs{
+			SubnetId:     privSubnet1.ID(),
+			RouteTableId: privateRouteTable.ID(),
+		})
+		if err != nil {
+			return err
+		}
+
+		_, err = ec2.NewRouteTableAssociation(ctx, prefix+"-rtb-assoc-priv-2", &ec2.RouteTableAssociationArgs{
+			SubnetId:     privSubnet2.ID(),
+			RouteTableId: privateRouteTable.ID(),
+		})
+		if err != nil {
+			return err
+		}
+
+		_, err = ec2.NewRouteTableAssociation(ctx, prefix+"-rtb-assoc-priv-3", &ec2.RouteTableAssociationArgs{
+			SubnetId:     privSubnet3.ID(),
+			RouteTableId: privateRouteTable.ID(),
+		})
+		if err != nil {
+			return err
+		}
+
+		// Associate Public Subs with Public Route Tables
+		_, err = ec2.NewRouteTableAssociation(ctx, prefix+"-rtb-assoc-pub-1", &ec2.RouteTableAssociationArgs{
+			SubnetId:     pubSubnet1.ID(),
+			RouteTableId: publicRouteTable.ID(),
+		})
+		if err != nil {
+			return err
+		}
+
+		_, err = ec2.NewRouteTableAssociation(ctx, prefix+"-rtb-assoc-pub-2", &ec2.RouteTableAssociationArgs{
+			SubnetId:     pubSubnet2.ID(),
+			RouteTableId: publicRouteTable.ID(),
+		})
+		if err != nil {
+			return err
+		}
+
+		_, err = ec2.NewRouteTableAssociation(ctx, prefix+"-rtb-assoc-pub-3", &ec2.RouteTableAssociationArgs{
+			SubnetId:     pubSubnet3.ID(),
+			RouteTableId: publicRouteTable.ID(),
+		})
+		if err != nil {
+			return err
+		}
+
+		// Resource: IAM Role
+		// Purpose: An IAM role is an IAM identity that you can create in your account that has specific permissions.
+		// Docs: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html
+
+		// IAM Role for EKS
+		eksRole, err := iam.NewRole(ctx, prefix+"eks-iam-eksRole", &iam.RoleArgs{
+			AssumeRolePolicy: pulumi.String(`{
```

Comment:
I think it's useful to move these policies elsewhere since it's a string possibly a constant? However, it's only used once and the other examples of this on the interweb appear to have it inline.